File: 141-LinkedListCycle/141-LinkedListCycle.py
# Last updated: 7/24/2025, 8:33:45 AM
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        slow,fast = head,head
        if not head:
            return False
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
            if fast == slow:
                return True
        return False

File: 141-LinkedListCycle/141-LinkedListCycle.cpp
// Last updated: 7/24/2025, 8:33:49 AM
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ios::sync_with_stdio(0);
        ListNode *ptr = head;
        while(ptr){
            if(ptr->val == 100001)
                return true;
            ptr->val = 100001;
            ptr = ptr->next;
        }
        return false;
    }
};